src/libraries/System.Threading.Tasks/tests/Task/TaskAPMTest.cs

```diff
@@ -89,6 +89,33 @@ public void PollUntilCompleteTechnique(bool hasReturnType)
             Assert.False(asyncResult.CompletedSynchronously, "Should not have completed synchronously.");
         }
 
+        [Theory]
+        [OuterLoop]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void PollUntilCompleteTechnique_TimeSpan(bool hasReturnType)
+        {
+            _hasReturnType = hasReturnType;
+
+            LongTask longTask;
+            if (_hasReturnType)
+                longTask = new LongTask<int>(LongTaskMilliseconds);
+            else
+                longTask = new LongTask(LongTaskMilliseconds);
+
+            IAsyncResult asyncResult = longTask.BeginDoTask(null, null);
+            var mres = new ManualResetEventSlim();
+
+            TimeSpan timeout = TimeSpan.FromMilliseconds(1);
+            while (!asyncResult.IsCompleted)
+            {
+                mres.Wait(timeout);
```

Comment:
this is testing ManualResetEVentSlim.Wait(TimeSpan) which already exists and is already hopefully tested.